src/ansiblelint/rules/no_jinja_nesting.md

```diff
@@ -0,0 +1,35 @@
+# no-jinja-nesting
+
+This rule checks for nested Jinja expressions.
+Avoid nesting Jinja expressions with curly brackets `{{ }}` inside other Jinja expressions.
+
+An Ansible rule is that "moustaches don't stack".
+Nested expressions are an Ansible anti-pattern and do not produce expected results.
+
+## Problematic Code
+
+```yaml
+---
+{{ list_one + {{ list_two | max }} }}
+```
+
+```yaml
+---
+{{ somevar_{{other_var}} }}
+```
+
+## Correct Code
+
+```yaml
+---
+{ { list_one + max(list_two) } }
+```
+
+```yaml
+---
+# Uses the hostvars variable.
+{{ hostvars[inventory_hostname]['somevar_' ~ other_var] }}
+
+# Uses the ansible.builtin.vars lookup plugin.
+{{ lookup('vars', 'somevar_' ~ other_var) }}
+```
```

Comment:
@ssbarnea One other question I have in this PR is about contextualizing these examples into playbooks. I found this one a little difficult and worry any playbook example might be nonsense. I'm still fairly new to Ansible and Jinja. Do we have "advanced" examples anywhere?

Reply:
While looking at this rule I realised that I will probably remove this rule by encapsulating the feature it provides into `jinja` one.

I will keep this PR open until I have the proof that I can bring that functionality into the jinja rule, especially as that would make it much more trustable (avoiding the regex, which brings false positivies)